lib/nats.js

```diff
@@ -713,6 +713,11 @@ Client.prototype.setupHandlers = function () {
     return
   }
 
+  if (this.connectionTimeoutHandler) {
+    clearTimeout(this.connectionTimeoutHandler)
+    delete this.connectionTimeoutHandler
```

Comment:
Does that make the test above fail? Meaning is that equivalent of = null?

Reply:
> Does that make the test above fail? Meaning is that equivalent of = null?

clear cancels the timeout - the issue was that one of the timers was left running and when the tls upgrade happened it would create a new timer. The new timer would be fine, but the old-timer would trigger. The subtle thing is on line 723 below, which is that when the timeout function fired, it destroyed the non-tls stream, so that left it in an inconsistent state (close was not raised on the socket, so it wouldn't reactivate reconnect).

Reply:
The test is effectively waiting longer than the timeout, and then chops the connection at the socket level, we are just wanting to see the reconnect happen at that point.

Reply:
Should we set the conncetionTimeoutHandler to nul?

Reply:
the delete does that